dbcrossbarlib/src/drivers/postgres_shared/catalog.rs

```diff
@@ -73,32 +64,51 @@ impl PgColumnSchema {
 /// Fetch information about a table from the database.
 ///
 /// Returns `None` if no matching table exists.
-pub(crate) fn fetch_from_url(
+pub(crate) async fn fetch_from_url(
+    ctx: &Context,
     database_url: &UrlWithHiddenPassword,
     table_name: &TableName,
 ) -> Result<Option<PgCreateTable>> {
-    let conn = PgConnection::establish(database_url.with_password().as_str())
-        .context("error connecting to PostgreSQL")?;
-
+    let client = connect(ctx, database_url).await?;
     let schema = table_name.schema().unwrap_or("public");
     let table = table_name.table();
 
     // Check to see if we have a table with this name.
-    let table_count = tables::table
-        .select(count_star())
-        .filter(tables::table_schema.eq(schema))
-        .filter(tables::table_name.eq(table))
-        .first::<i64>(&conn)?;
+    let count_matching_tables_sql = r#"
+SELECT COUNT(*) AS count
+FROM information_schema.tables
+WHERE
+    table_schema = $1 AND
+    table_name = $2
+"#;
+    let row = client
+        .query_one(count_matching_tables_sql, &[&schema, &table])
+        .await?;
```

Comment:
very satisfying

at this point in my career i'm just... not very interested in sql generation

didn't you mention there was a rust postgres driver that would parse the sql you provide and get types from it?

Reply:
Yes, `sqlx`.

`diesel` is really an excellent example of what can be done with a sufficiently smart generator, and it works beautifully in something like `falconeri` where we control the tables & migrations. It ensures that the Rust code and our queries actually match the database, in detail. That's useful because `falconeri` doesn't have enough integration tests, bec

But here we only have like 3 tables, none of which we control, so it's not buying us anywhere _near_ enough to put up with two Postgres drivers. I have no plans to switch from `tokio-postgres` to `sqlx`, because we have plenty of experience doing giant `COPY` operations with `tokio-postgres`, and because `sqlx` wouldn't buy us much here for reasons similar to `diesel`. Besides, this code changes once a year or so. Getting strong types for just these three queries is not a priority.